internal/services/automation/automation_schedule_resource.go

```diff
@@ -263,7 +263,11 @@ func resourceAutomationScheduleCreateUpdate(d *pluginsdk.ResourceData, meta inte
 
 	if v, ok := d.GetOk("expiry_time"); ok {
 		t, _ := time.Parse(time.RFC3339, v.(string)) // should be validated by the schema
-		parameters.Properties.SetExpiryTimeAsTime(t.In(loc))
+		// fixes: https://github.com/hashicorp/terraform-provider-azurerm/issues/21854. that year 9999 may return by API
+		if t.In(loc).Year() <= 9999 {
+			t = t.In(loc)
+		}
```

Comment:
yes I think it can always send a UTC time here. but I am not sure if it will cause changes to current users if we change to UTC here.

 the timezone logic introduced in [this PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/20568/files#diff-0fd1fe814099ff079f86fb5402d66cdd2fdc77b6c93bd2b90adef95a5526b65dL258) if @catriona-m has any context about this logic and can we change the `loc` to UTC safely?

Reply:
if we have a user-specified value here, we would still need it to be in the timezone they have specified too I think, so we do need to check if it's the default value from the api or not

Reply:
@tombuildsstuff Is this PR acceptable to resolve the issue, or do you prefer @catriona-m's suggestion to avoid setting expiry_time as a timestamp in the Read in first place if it hasn't been specified? both options seem viable to me.

Reply:
actually we don't need to do either approach, we can instead set the raw string value rather than parsing and then setting the date time?

Reply:
that should work too. I'll update it that way.

Reply:
PR updated and added a new acc test for it:

```
--- PASS: TestAccAutomationSchedule_oneTime_basic (158.45s)
--- PASS: TestAccAutomationSchedule_expiryTimeOfEuropeTimeZone (195.06s)
--- PASS: TestAccAutomationSchedule_requiresImport (166.47s)
--- PASS: TestAccAutomationSchedule_oneTime_complete (95.21s)
--- PASS: TestAccAutomationSchedule_oneTime_update (126.64s)
--- PASS: TestAccAutomationSchedule_hourly (91.86s)
--- PASS: TestAccAutomationSchedule_daily (94.86s)
--- PASS: TestAccAutomationSchedule_weekly (94.92s)
--- PASS: TestAccAutomationSchedule_monthly (154.84s)
--- PASS: TestAccAutomationSchedule_weekly_advanced (94.75s)
--- PASS: TestAccAutomationSchedule_monthly_advanced_by_day (92.64s)
--- PASS: TestAccAutomationSchedule_monthly_advanced_by_week_day (93.93s)
```